test/functional/change_stream_spec_tests.js

```diff
@@ -0,0 +1,258 @@
+'use strict';
+
+const EJSON = require('mongodb-extjson');
+const chai = require('chai');
+const fs = require('fs');
+const camelCase = require('lodash.camelcase');
+const MongoClient = require('../../lib/mongo_client');
+const setupDatabase = require('./shared').setupDatabase;
+const delay = require('./shared').delay;
+const expect = chai.expect;
+
+describe('Change Stream Spec', function() {
+  const EJSONToJSON = x => JSON.parse(EJSON.stringify(x));
+
+  before(function() {
+    return setupDatabase(this.configuration).then(() => {
+      this.globalClient = new MongoClient(this.configuration.url());
+      return this.globalClient.connect();
+    });
+  });
+
+  after(function() {
+    const gc = this.globalClient;
+    this.globalClient = undefined;
+    return new Promise(r => gc.close(() => r()));
+  });
+
+  fs
+    .readdirSync(`${__dirname}/spec/change-stream`)
+    .filter(filename => filename.match(/\.json$/))
+    .forEach(filename => {
+      const specString = fs.readFileSync(`${__dirname}/spec/change-stream/${filename}`, 'utf8');
+      const specData = JSON.parse(specString);
+
+      const ALL_DBS = [specData.database_name, specData.database2_name];
+
+      describe(filename, () => {
+        beforeEach(function() {
+          const gc = this.globalClient;
+          const sDB = specData.database_name;
+          const sColl = specData.collection_name;
+          return Promise.all(ALL_DBS.map(db => gc.db(db).dropDatabase()))
+            .then(() => gc.db(sDB).createCollection(sColl))
+            .then(() =>
+              new MongoClient(this.configuration.url(), { monitorCommands: true }).connect()
+            )
+            .then(client => {
+              const ctx = (this.ctx = {});
+              const events = (this.events = []);
+              ctx.gc = gc;
+              ctx.client = client;
+              ctx.database = ctx.client.db(sDB);
+              ctx.collection = ctx.database.collection(sColl);
+              ctx.client.on('commandStarted', e => events.push(e));
+            });
+        });
+
+        afterEach(function(done) {
+          const ctx = this.ctx;
+          this.ctx = undefined;
+          this.events = undefined;
+
+          if (ctx.client) {
+            ctx.client.close(e => done(e));
```

Comment:
nit: you can just return the promise, or do nothing, and remove the need for `done`